genesis/utils/gltf.py

```diff
@@ -0,0 +1,390 @@
+        opacity_texture = color_texture.check_dim(3)
+        if opacity_texture is not None:
+            opacity_texture.apply_cutoff(alpha_cutoff)
+
+    if "KHR_materials_unlit" in material.extensions:
+        # No unlit material implemented in renderers. Use emissive texture.
+        if color_texture is not None:
+            emissive_texture = color_texture
+            color_texture = None
+        material.extensions.pop("KHR_materials_unlit")
+    else:
+        # parse emissive
+        emissive_image = None
+        if material.emissiveTexture is not None:
+            texture = glb.textures[material.emissiveTexture.index]
+            if material.emissiveTexture.texCoord is not None:
+                uvs_used = material.emissiveTexture.texCoord
+            emissive_image = get_glb_image(texture.source, "RGB")
+
+        emissive_factor = None
+        if material.emissiveFactor is not None:
+            emissive_factor = np.array(material.emissiveFactor, dtype=float)
+
+        if emissive_factor is not None and np.any(emissive_factor > 0.0):  # Make sure to check emissive
+            emissive_texture = mu.create_texture(emissive_image, emissive_factor, "srgb")
+
+    # TODO: Parse them!
+    for extension_name, extension_material in material.extensions.items():
+        if extension_name == "KHR_materials_specular":
+            specular_weight = extension_material.get("specularFactor", 1.0)
+            specular_color = np.array(extension_material.get("specularColorFactor", [1.0, 1.0, 1.0]), dtype=float)
+
+        elif extension_name == "KHR_materials_clearcoat":
+            clearcoat_weight = extension_material.get("clearcoatFactor", 0.0)
+            clearcoat_roughness_factor = (extension_material["clearcoatRoughnessFactor"],)
+
+        elif extension_name == "KHR_materials_volume":
+            attenuation_distance = extension_material["attenuationDistance"]
+
+        elif extension_name == "KHR_materials_transmission":
+            specular_trans_factor = extension_material.get("transmissionFactor", 0.0)  # e.g. 1
+
+        elif extension_name == "KHR_materials_ior":
+            ior = extension_material["ior"]  # e.g. 1.4500000476837158
+
+    material_surface = surface.copy()
+    material_surface.update_texture(
+        color_texture=color_texture,
+        opacity_texture=opacity_texture,
+        roughness_texture=roughness_texture,
+        metallic_texture=metallic_texture,
+        normal_texture=normal_texture,
+        emissive_texture=emissive_texture,
+        ior=ior,
+        double_sided=double_sided,
+    )
+
+    return material_surface, uvs_used
+
+
+def parse_glb_tree(glb, node_index):
+    node = glb.nodes[node_index]
+    if node.matrix is not None:
+        matrix = np.array(node.matrix, dtype=float).reshape((4, 4))
+    else:
+        matrix = np.identity(4, dtype=float)
+        if node.translation is not None:
+            translation = np.array(node.translation, dtype=float)
```

Comment:
Don't do math for this. You are multiplying by identity. Just fill the matrix already:
```python
matrix[3, :3] = translation
```